cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/YBSnapshotWithPgSinkIT.java

```diff
@@ -0,0 +1,323 @@
+        sinkConfig = pgHelper.getJdbcSinkConfiguration(postgresContainer, "id");
+        kafkaConnectContainer.registerConnector(SINK_CONNECTOR_NAME, sinkConfig);
+    }
+
+    @AfterEach
+    public void afterEachTest() throws Exception {
+        // Assuming that CDCSDK Server container is started in each test, we will stop/kill it here
+        cdcsdkContainer.stop();
+
+        // Deregister the sinkConnector
+        kafkaConnectContainer.deleteConnector(SINK_CONNECTOR_NAME);
+
+        // Drop the source and sink tables now
+        dropTablesAfterEachTest(DEFAULT_TABLE_NAME);
+
+        // Delete the topic in Kafka as well so that it does not contain irrelevant data for other tests
+        kafkaHelper.deleteTopicInKafka(ybHelper.getKafkaTopicName());
+    }
+
+    @AfterAll
+    public static void afterAll() throws Exception {
+        // Stop the running containers
+        postgresContainer.stop();
+        kafkaConnectContainer.stop();
+        kafkaContainer.stop();
+    }
+
+    @Test
+    public void verifyBasicSnapshotStreaming() throws Exception {
+        int rowsToBeInserted = 100;
+        // Insert data to the source table
+        TestHelper.insertRowsInSourceTable(ybHelper, DEFAULT_TABLE_NAME, rowsToBeInserted);
+
+        // Start the CDCSDKContainer with initial snapshot mode
+        cdcsdkContainer = kafkaHelper.getCdcsdkContainer(ybHelper, "public." + DEFAULT_TABLE_NAME, 10, "initial");
+        cdcsdkContainer.withNetwork(containerNetwork);
+        cdcsdkContainer.start();
+
+        pgHelper.waitTillRecordsAreVerified(rowsToBeInserted, 10000);
+
+        pgHelper.assertRecordCountInPostgres(rowsToBeInserted);
+    }
+
+    @Test
+    public void performStreamingAfterSnapshotAndPerformInsUpdDelOps() throws Exception {
+        int rowsToBeInserted = 100;
+        // Insert data to the source table
+        TestHelper.insertRowsInSourceTable(ybHelper, DEFAULT_TABLE_NAME, rowsToBeInserted);
+
+        // Start the CDCSDKContainer with initial snapshot mode
+        cdcsdkContainer = kafkaHelper.getCdcsdkContainer(ybHelper, "public." + DEFAULT_TABLE_NAME, 10, "initial");
+        cdcsdkContainer.withNetwork(containerNetwork);
+        cdcsdkContainer.start();
+
+        pgHelper.waitTillRecordsAreVerified(rowsToBeInserted, 10000);
+
+        // Now stop the cdcsdk container and start it again with never snapshot mode
+        cdcsdkContainer.stop();
+
+        // Change the snapshot mode and start the container
+        cdcsdkContainer.withEnv("CDCSDK_SOURCE_SNAPSHOT_MODE", "never");
+        cdcsdkContainer.start();
+
+        for (int i = 100; i < 150; ++i) {
+            ybHelper.execute(UtilStrings.getInsertStmt(DEFAULT_TABLE_NAME, i, "first_" + i, "last_" + i, 23.45));
+        }
+
+        pgHelper.waitTillRecordsAreVerified(150, 10000);
+        pgHelper.assertRecordCountInPostgres(150);
+
+        ybHelper.execute("UPDATE " + DEFAULT_TABLE_NAME + " SET first_name = 'Vaibhav' WHERE id >= 100 AND id <= 149;");
```

Comment:
Should we check if only last 50 records were streamed?

Reply:
I am verifying the updated data in the sink just after this line in the Awaitility block. Do I need to also verify those records in Kafka?